### PR TITLE
removed code climate from go orb

### PR DESCRIPTION
**Pull request by jriegner**
**What this PR does / why we need it**:

This PR removes code climate from the go build pipeline. Since we are not using it anymore and especially in bidder project, the repo is already removed on CC side which blocks the build of bidder.

**Special notes for your reviewer**:

**If applicable**:
- [x] All new Jobs, Commands, Executors, Parameters have descriptions
- [x] If PR adds a new feature, Examples have been added
- [x] README has been updated, if necessary

### CODE REVIEW

**Review by bpietraga — DISMISSED** [review]
👍

File: src/go/jobs/test_and_coverage.yaml [hunk 1]
```diff
@@ -66,9 +64,4 @@ steps:
         cat cover-source.out | grep -v '.pb.go' | grep -v '.pqt.go' | grep -v '.mock.go' > cover-final.out
         go tool cover -func=cover-final.out
```
Comment (ders) [inline src/go/jobs/test_and_coverage.yaml:65]: Does this mean that we still have a coverage report?  If so, where will we see it?

File: src/go/jobs/test_and_coverage.yaml [hunk 2]
```diff
@@ -66,9 +64,4 @@ steps:
         cat cover-source.out | grep -v '.pb.go' | grep -v '.pqt.go' | grep -v '.mock.go' > cover-final.out
```
Comment (ders) [inline src/go/jobs/test_and_coverage.yaml:64]: Not really related to this PR, but how about removing pqt, as it's a long-gone Piotr artifact?

Comment (jriegner) [reply src/go/jobs/test_and_coverage.yaml:64]: Might be this qtypes thingy no? 🤔 No clue but I can have a look

Comment (ders) [reply src/go/jobs/test_and_coverage.yaml:64]: Just need to take out the `grep -v '.pqt.go'` step (and update the comment).

**Comment by ders** [comment]
Can lines 39-49 of coverage.yaml also be removed?

File: src/go/jobs/test_and_coverage.yaml (hunk 2 above)
Comment (ders) [reply src/go/jobs/test_and_coverage.yaml]: Doh.  This line is removed anyway.  Nevermind.

**Review by ders — DISMISSED** [review]

**Review by cmoroianu1 — DISMISSED** [review]
LGTM

**Review by ders — APPROVED** [review]

**Pull request merged**